File: arrays/contains-duplicate/duplicate.py
def contains_duplicate(nums):
  hash_map = dict({})
  
  for num in nums:
    curr_val = hash_map.get(num)
    if curr_val is None:
      hash_map[num] = 1
    else:
      return True
  
  return False

nums = [1,2,3,4,1]

print(contains_duplicate(nums))

File: arrays/contains-duplicate/duplicate.go
package main

import "fmt"

func containsDuplicate(nums []int) bool {

	hash_map := make(map[int]int)

	for i := 0; i < len(nums); i++ {
		currVal, exists := hash_map[nums[i]]
		if exists {
			return true
		} else {
			hash_map[nums[i]] = currVal + 1
		}
	}
	return false
}

func main() {
	list := []int{1, 2, 3, 4, 5}
	fmt.Println(containsDuplicate((list)))
}
